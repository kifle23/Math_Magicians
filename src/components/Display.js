import PropTypes from 'prop-types';

function Display({ total, operation, next }) {
  return (
    <div className="display">
      {total}
      {operation}
      {next}
    </div>
  );
}

Display.propTypes = {
  total: PropTypes.number.isRequired,
  operation: PropTypes.string,
  next: PropTypes.number,
};
Display.defaultProps = {
  operation: null,
  next: null,
};

export default Display;
