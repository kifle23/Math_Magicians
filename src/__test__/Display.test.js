import React from 'react';
import renderer from 'react-test-renderer';
import Display from '../components/Display';

describe('Display', () => {
  test('renders correctly with props', () => {
    const tree = renderer
      .create(<Display total={10} operation="+" next="5" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with default props', () => {
    const tree = renderer.create(<Display />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('does not render total when it is 0', () => {
    const tree = renderer.create(<Display total={0} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
