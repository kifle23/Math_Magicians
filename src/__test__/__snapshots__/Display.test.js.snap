// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display does not render total when it is 0 1`] = `
<div
  className="display"
/>
`;

exports[`Display renders correctly with default props 1`] = `
<div
  className="display"
/>
`;

exports[`Display renders correctly with props 1`] = `
<div
  className="display"
>
  10
  +
  5
</div>
`;
